Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -244,6 +259,13 @@ extension HTTPConnectionPool {
         mutating func waitingForConnectivity(_ error: Error, connectionID: Connection.ID) -> Action {
             self.lastConnectFailure = error
 
+            guard self.retryConnectionEstablishment else {
+                return .init(
+                    request: self.failAllRequests(reason: error),
+                    connection: .none
+                )
+            }
+
```

Comment:
This is weird! If we always fail all requests, when we a connection attempt failed, we should never reschedule a connection attempt. And this means we should never reach this point.

Reply:
`waitingForConnectivity` is called if a `NWConnection` transitions into the waiting state which has nothing to do with our own connection retry logic. This means that it will be called if a NWConnection "fails" instead of `failedToCreateNewConnection` because the connection will do its own retry thing.

Reply:
Can we please add a code comment, that this is only triggered in the `Network.framework` flow?

If we don't want to retry connection establishment, NW.framework should not go into the waiting state, should it?

Reply:
I have removed the code again. We have a different flag (`networkFrameworkWaitForConnectivity`) for disabling Network.frameworks waiting behaviour.